beacon-chain/rpc/prysm/validator/validator_count_test.go

```diff
@@ -18,6 +26,106 @@ import (
 	"github.com/prysmaticlabs/prysm/v4/testing/util"
 )
 
+// MockStater is a fake implementation of lookup.Stater.
+// Note: This MockStater ensures the provided state identifier is correct.
+type MockStater struct {
```

Comment:
Would it be possible to extend `beacon-chain/rpc/testutil/mock_stater.go` rather than creating a new mock?

Reply:
sure, I have added state id logic into the existing stater.